sources/constraint-declaration-validation.asciidoc

```diff
@@ -1573,7 +1930,184 @@ The resolutions shown in <<table-constraintvalidator-resolution>> occur.
+* If no `ValueExtractor` type-compliant and container-element-compliant with `C` is found among the available value extractors, a `ConstraintDeclarationException` is raised.
+* A `ValueExtractor<U>` type-compliant with `C` is considered _strictly more specific_ than a `ValueExtractor<V>` compliant with `C` if `U` is a strict subtype of `V`.
+`U` is a strict subtype of `V` if `U` is a subtype of `V` and `U` != `V`.
+* A `ValueExtractor<U>` type-compliant with `C` is considered maximally specific if no other `ValueExtractor<V>` type-compliant with `C` is strictly more specific than `ValueExtractor<U>`.
+* If more than one maximally specific and container-element-compliant `ValueExtractor` is found, a `ConstraintDeclarationException` is raised.
+
+[NOTE]
+.Implementation note
+====
+Extractor retrieval for container element constraints is based on the declared type of constrained elements,
+hence it is recommended that implementations perform the resolution once and then cache the value extractor for a given constraint.
+====
+
+[[constraintdeclarationvalidationprocess-validationroutine-valueextractorresolution-algorithm-cascaded]]
+===== ValueExtractor resolution algorithm for cascaded validation
+
+For a container with the declared type `C` and the runtime type `C'` whose element type is marked for cascaded validation, the following resolution rules apply for identifying the value extractor:
+
+* A `ValueExtractor<T>` is said to be _type-compliant_ with `C'`, if `C'` is a subtype of `T`.
+Note that `C'` is a subtype of `T` if `C'` = `T`.
+* A `ValueExtractor` implementation is said to be _container-element-compliant_ with `C`, if it handles a type parameter that maps to the type argument marked with `@Valid`.
+* If no `ValueExtractor` type-compliant with `C'` and container-element-compliant with `C` is found among the available value extractors, a `ConstraintDeclarationException` is raised.
+* A `ValueExtractor<U>` type-compliant with `C'` is considered _strictly more specific_ than a `ValueExtractor<V>` compliant with `C'` if `U` is a strict subtype of `V`.
+`U` is a strict subtype of `V` if `U` is a subtype of `V` and `U` != `V`.
+* A `ValueExtractor<U>` type-compliant with `C'` is considered maximally specific if no other `ValueExtractor<V>` type-compliant with `C'` is strictly more specific than `ValueExtractor<U>`.
+* If more than one maximally specific and container-element-compliant `ValueExtractor` is found, a `ConstraintDeclarationException` is raised.
+
+[NOTE]
+.Implementation note
+====
+Extractor retrieval for container element constraints is based on the runtime type of the elements marked with `@Valid`
+(this is to be consistent with the semantics of property paths for cascaded validation defined in Bean Validation 1.x),
```

Comment:
People reading the spec don't care that things are from 1.x. so s/ defined in Bean Validation 1.x//